app/src/main/java/org/kiwix/kiwixmobile/wifi_hotspot/HotspotService.java

```diff
@@ -0,0 +1,219 @@
+package org.kiwix.kiwixmobile.wifi_hotspot;
+
+import android.app.Notification;
+import android.app.NotificationChannel;
+import android.app.NotificationManager;
+import android.app.PendingIntent;
+import android.app.Service;
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.content.IntentFilter;
+import android.net.wifi.WifiConfiguration;
+import android.os.Binder;
+import android.os.Build;
+import android.os.IBinder;
+import android.util.Log;
+import android.widget.Toast;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.core.app.NotificationCompat;
+import org.kiwix.kiwixmobile.R;
+import org.kiwix.kiwixmobile.utils.Constants;
+import org.kiwix.kiwixmobile.utils.ServerUtils;
+import org.kiwix.kiwixmobile.webserver.ZimHostCallbacks;
+import org.kiwix.kiwixmobile.webserver.WebServerHelper;
+import org.kiwix.kiwixmobile.webserver.ZimHostActivity;
+
+import static org.kiwix.kiwixmobile.webserver.ZimHostActivity.SELECTED_ZIM_PATHS_KEY;
+
+/**
+ * HotspotService is used to add a foreground service for the wifi hotspot.
+ * Created by Adeel Zafar on 07/01/2019.
+ */
+
+public class HotspotService extends Service implements HotspotStateListener {
+
+  public static final String ACTION_TURN_ON_AFTER_O = "Turn_on_hotspot_after_oreo";
+  public static final String ACTION_TURN_OFF_AFTER_O = "Turn_off_hotspot_after_oreo";
+  public static final String ACTION_IS_HOTSPOT_ENABLED = "Is_hotspot_enabled";
+  public static final String ACTION_START_SERVER = "start_server";
+  public static final String ACTION_STOP_SERVER = "stop_server";
+
+  private static final int HOTSPOT_NOTIFICATION_ID = 666;
+  private static final String ACTION_STOP = "hotspot_stop";
+  private static final String TAG = "HotspotService";
+  private WifiHotspotManager hotspotManager;
+  private BroadcastReceiver stopReceiver;
+  private NotificationManager notificationManager;
+  private NotificationCompat.Builder builder;
+  private ZimHostCallbacks zimHostCallbacks;
+  private final IBinder serviceBinder = new HotspotBinder();
+  private WebServerHelper webServerHelper;
+
+  @Override public void onCreate() {
+
+    super.onCreate();
+
+    hotspotManager = new WifiHotspotManager(this);
+    hotspotManager.registerListener(HotspotService.this);
```

Comment:
you don't need to qualify the `this` with HotspotService. Only qualify `this` if it really lends to readability or is actually necessary.

Reply:
If I remove `this` It throws `Expression expected`.

Reply:
Fixed. ad2537e7b904c8994678928c2603a5c92a457ef0